plugins/auth-backend/src/providers/oidc/provider.ts

```diff
@@ -14,269 +14,56 @@
-        this.resolverContext,
-      );
-    }
-
-    return {
-      backstageIdentity,
-      providerInfo: {
-        idToken: result.tokenset.id_token,
-        accessToken: result.tokenset.access_token!,
-        scope: result.tokenset.scope!,
-        expiresInSeconds,
-      },
-      profile,
-    };
-  }
-}
-
 /**
  * Auth provider integration for generic OpenID Connect auth
  *
  * @public
  */
 export const oidc = createAuthProviderIntegration({
   create(options?: {
-    authHandler?: AuthHandler<OidcAuthResult>;
+    /**
+     * The profile transformation function used to verify and convert the auth response
+     * into the profile that will be presented to the user.
+     */
+    authHandler?: AuthHandler<OAuthResult>;
 
+    /**
+     * Configure sign-in for this provider, without it the provider can not be used to sign users in.
+     */
     signIn?: {
-      resolver: SignInResolver<OidcAuthResult>;
+      resolver: SignInResolver<OAuthResult>;
     };
   }) {
-    return ({ providerId, globalConfig, config, resolverContext }) =>
-      OAuthEnvironmentHandler.mapConfig(config, envConfig => {
-        const clientId = envConfig.getString('clientId');
-        const clientSecret = envConfig.getString('clientSecret');
-        const customCallbackUrl = envConfig.getOptionalString('callbackUrl');
-        const callbackUrl =
-          customCallbackUrl ||
-          `${globalConfig.baseUrl}/${providerId}/handler/frame`;
-        const metadataUrl = envConfig.getString('metadataUrl');
-        const tokenEndpointAuthMethod = envConfig.getOptionalString(
-          'tokenEndpointAuthMethod',
-        ) as ClientAuthMethod;
-        const tokenSignedResponseAlg = envConfig.getOptionalString(
-          'tokenSignedResponseAlg',
-        );
-        const scope = envConfig.getOptionalString('scope');
-        const prompt = envConfig.getOptionalString('prompt');
-
-        const authHandler: AuthHandler<OidcAuthResult> = options?.authHandler
-          ? options.authHandler
-          : async ({ userinfo }) => ({
-              profile: {
-                displayName: userinfo.name,
-                email: userinfo.email,
-                picture: userinfo.picture,
-              },
-            });
-
-        const provider = new OidcAuthProvider({
-          clientId,
-          clientSecret,
-          callbackUrl,
-          tokenEndpointAuthMethod,
-          tokenSignedResponseAlg,
-          metadataUrl,
-          scope,
-          prompt,
-          signInResolver: options?.signIn?.resolver,
-          authHandler,
-          resolverContext,
-        });
-
-        return OAuthAdapter.fromConfig(globalConfig, provider, {
-          providerId,
-          callbackUrl,
-        });
-      });
+    return createOAuthProviderFactory({
+      authenticator: oidcAuthenticator,
+      profileTransform: adaptLegacyOAuthHandler(options?.authHandler),
+      signInResolver: adaptLegacyOAuthSignInResolver(options?.signIn?.resolver),
```

Comment:
Because of those type incompatibilities, it seems like we can't use the `adaptLegacy*` helpers here

Reply:
removed the use of these helpers in fd32e0d